src/rpcq.lisp

```diff
@@ -573,38 +282,95 @@ We distinguish between the following options for any field type:
-               ,@(when documentation
-                   `((:documentation ,(format-documentation-string documentation)))))
-             
-             (defmethod %serialize ((,obj ,class-name))
-               (with-slots ,slot-names ,obj
-                 (let ((,hash-table (make-hash-table :test #'equal)))
-                   (setf (gethash "_type" ,hash-table) ,(symbol-name class-name))
-                   ,@(loop :for slot :in slot-names
-                           :collect `(setf (gethash ,(symbol-name slot) ,hash-table)
-                                           (%serialize ,slot)))
-                   ,hash-table)))
-
-             (defmethod %deserialize-struct ((type (eql ',class-name)) (payload hash-table))
-               (assert (string= (gethash "_type" payload) ,(symbol-name class-name)))
-               (make-instance ',class-name
-                              ,@(mapcan (init-spec 'payload) slot-names)))
-
-
-             (defmethod print-object ((,obj ,class-name) stream)
-               (print-unreadable-object (,obj stream :type t)
-                 (pprint-indent :block 2)
-                 ,@(loop :for slot :in slot-names
-                         :collect `(format stream
-                                           "~&  ~A -> ~S"
-                                           ,(symbol-name slot)
-                                           (,(accessor slot) ,obj)))))))))))
+    (alexandria:with-gensyms (obj init-args)
+      (let ((slot-names (mapcar #'car field-specs)))
+        `(progn
+           (defclass ,class-name ,parent-name
+             ,(mapcar #'make-slot-spec field-specs)
+             ,@(when documentation
+                 `((:documentation ,(format-documentation-string documentation)))))
+           
+           (defmethod shared-initialize ((instance ,class-name) slot-names
+                                           &rest initargs
+                                           &key &allow-other-keys)
+               ,@(mapcan #'slot-initialization-deprecation-warnings field-specs)
+               (call-next-method))
+           
+           ,@(mapcan #'slot-accessor-deprecation-warnings field-specs)
+           
+           (defmethod %serialize ((,obj ,class-name))
+             (%%serialize ,obj (make-hash-table :test #'equal)))
+           
+           (defmethod %%serialize ((,obj ,class-name) (hash-table hash-table))
+             (with-slots ,slot-names ,obj
+               ,@(loop :for slot :in slot-names
+                       :collect `(setf (gethash ,(symbol-name slot) hash-table)
+                                       (%serialize ,slot)))
+               (call-next-method)
+               (setf (gethash "_type" hash-table) ,(symbol-name class-name))
+               hash-table))
+
+           (defmethod %deserialize-struct ((type (eql ',class-name)) (payload hash-table))
+             (assert (string= (gethash "_type" payload) ,(symbol-name class-name)))
```

Comment:
u will be please to know that `string=` works with string designators, so `(string= 'x "X")` returns `T`.